runtime/cudaq/platform/qpu.h

```diff
@@ -174,8 +174,17 @@ class QPU : public registry::RegisteredType<QPU> {
   /// as a struct-packed void pointer and its corresponding size.
   virtual void launchKernel(const std::string &name, void (*kernelFunc)(void *),
                             void *args, std::uint64_t, std::uint64_t) = 0;
+
+  /// Launch the kernel with given name and argument arrays.
+  // This is intended for remote QPUs whereby we need to JIT-compile the kernel
+  // with argument synthesis. Remote QPU implementation to override this.
   virtual void launchKernel(const std::string &name,
-                            const std::vector<void *> &rawArgs) = 0;
+                            const std::vector<void *> &rawArgs) {
+    if (!isRemote())
+      throw std::runtime_error("Wrong kernel launch point: Attempt to launch "
+                               "kernel in JIT mode on local "
```

Comment:
The documentation calls this "streamlined for JIT". Can we add that phrasing here?

Reply:
Sure. Fixed in https://github.com/NVIDIA/cuda-quantum/pull/2111/commits/83b67ba456c22704dd11a56b65bb7ca1b2c27ff7